From patchwork Thu Apr 11 18:01:46 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Nadav Amit <nadav.amit@gmail.com>
X-Patchwork-Id: 10896579
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id F40581390
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 11 Apr 2019 18:01:58 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E276F28D84
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 11 Apr 2019 18:01:58 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id D641A28D97; Thu, 11 Apr 2019 18:01:58 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 83BE928D88
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 11 Apr 2019 18:01:58 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726836AbfDKSB5 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 11 Apr 2019 14:01:57 -0400
Received: from mail-wr1-f68.google.com ([209.85.221.68]:42876 "EHLO
        mail-wr1-f68.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726615AbfDKSB4 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 11 Apr 2019 14:01:56 -0400
Received: by mail-wr1-f68.google.com with SMTP id g3so8515686wrx.9
        for <kvm@vger.kernel.org>; Thu, 11 Apr 2019 11:01:55 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=9LI8Xe2X6jJ4shS0/q2sFCzTGRz3alxOPmz9fWaIJrE=;
        b=ukuZ3PpNVpN/Ps0V4gYaM/ov2Lp1S8eg6Oym3z4G+6R5+CBI7/eJ446gczwJOFBDjM
         Oi8yaO0QY/BvwwfsgiTPggazZmXYUMqxfrYllAPKWbiYaVSZgLJkeC9s7iaP25FxL6jg
         Vmoi7E3TigMLlpZIEe73RY8C6tdyf/FhSsy+bKSpKY7b53tb/vmYC3ypjD1Z4OwxvdTB
         hn4ajZax6G5uGjZwNH5iqUnq8yxPDED/sYoJiUIJdRE5tfzUIqhNWkDwmua8+a/FBEU7
         2rt6fs069K3Rpcx4l/JA4sEpOl9vX5B8rY6S9CUX1IJ9vK/eml/ZMHwothfDNaW9kXB6
         c+Ow==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=9LI8Xe2X6jJ4shS0/q2sFCzTGRz3alxOPmz9fWaIJrE=;
        b=aShKm2N5aGU7+hJLth5p27nIhxERZ81O6XzDBRH0P7OQ6zUDdPmaNLo/E/G5ucVxtf
         ExvqrCiwX+pxYl8272THEtTL+/xQVJqJ3F8Qq4L1OBqZuK1ufRFueWPN5cV9fOFGVylF
         q29/A1ZPGbfo5xM8TsUqfD/JsK+xN5AGkkU/+lcjLJBdaizSRnyqrTI1KRphINjQUnIk
         i1BYLp7UQsJjSHGe1duEkQQPuRFnvj45GcU/JlLUqomaa14gI6V8u0ZeX3KPfRMGIxgP
         o4hvGypBNclrCC4W+4zy6OMJ3ShiDrN+PiMn7cJNaXLRwM0oevtS5D3hFQZ/stNkaWaC
         fzew==
X-Gm-Message-State: APjAAAXv1QnXcf5cDPn9KA0AiQRwpv4xhAZT+Ott9WH77AYudDvBPYbU
        EI38uHFYU0HaUeNxUPTso4uChvvQDDY=
X-Google-Smtp-Source: 
 APXvYqzHtle9oiEMUHk0ilZ1rQaJud6VH8vfsGFwFLugNJVM8L85KKiNOmAUh58u7tWlPcpd9NPrjA==
X-Received: by 2002:adf:e84b:: with SMTP id
 d11mr30886818wrn.289.1555005714807;
        Thu, 11 Apr 2019 11:01:54 -0700 (PDT)
Received: from htb-2n-eng-dhcp405.eng.vmware.com ([66.170.99.1])
        by smtp.gmail.com with ESMTPSA id
 m6sm48501308wrr.53.2019.04.11.11.01.52
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Thu, 11 Apr 2019 11:01:54 -0700 (PDT)
From: Nadav Amit <nadav.amit@gmail.com>
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>,
 kvm@vger.kernel.org, Nadav Amit <nadav.amit@gmail.com>,
 Sean Christopherson <sean.j.christopherson@intel.com>
Subject: [kvm-unit-tests PATCH v2] x86: Fix wrong test for 5-level page-table
Date: Thu, 11 Apr 2019 11:01:46 -0700
Message-Id: <20190411180146.24387-1-nadav.amit@gmail.com>
X-Mailer: git-send-email 2.19.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Untested, but clearly bit 12 is the one that needs to be tested.

Cc: Sean Christopherson <sean.j.christopherson@intel.com>
Signed-off-by: Nadav Amit <nadav.amit@gmail.com>
---
 x86/cstart64.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/cstart64.S b/x86/cstart64.S
index 21db10f..cccfcea 100644
--- a/x86/cstart64.S
+++ b/x86/cstart64.S
@@ -243,7 +243,7 @@ start64:
 setup_5level_page_table:
 	/* Check if 5-level paging has already enabled */
 	mov %cr4, %rax
-	test $12, %eax
+	test $0x1000, %eax
 	jnz lvl5
 
 	pushq $32
